packages/flutter/lib/src/animation/animation.dart

```diff
@@ -60,33 +60,93 @@ typedef AnimationStatusListener = void Function(AnimationStatus status);
+/// Consumers of the animation can listen for changes to either the value
+/// or the status, with [addListener] and [addStatusListener].
+/// The listener callbacks are called during the "animation" phase of
 /// the pipeline, just prior to rebuilding widgets.
 ///
-/// To create a new animation that you can run forward and backward, consider
-/// using [AnimationController].
+/// An animation may move forward or backward on its own as time passes
+/// (like the opacity of a button that fades over a fixed duration
+/// once the user touches it),
+/// or it may be driven by the user
+/// (like the position of a slider that the user can drag back and forth),
+/// or it may do both
+/// (like a switch that snaps into place when releaseed,
+/// or a [Dismissible] that responds to drag and fling gestures, etc.).
+/// The behavior is normally controlled by method calls on
+/// some underlying [AnimationController],
+/// for example by a [State] that created the [AnimationController].
+/// When an animation is actively animating, it typically updates on
+/// each frame, driven by a [Ticker].
+///
+/// ## Using animations
+///
+/// For simple animation effects, consider using one of the
+/// [ImplicitlyAnimatedWidget] subclasses,
+/// like [AnimatedScale], [AnimatedOpacity], and many others.
+/// When an [ImplicitlyAnimatedWidget] suffices, there is
+/// no need to work with [Animation], [AnimationController], or [Tween],
+/// or even to have a [State] to manage the animation (though all of those are
+/// involved in the implementation of the [ImplicitlyAnimatedWidget] itself).
+///
+/// Otherwise, typically an animation originates with an [AnimationController]
+/// (which is itself an [Animation<double>])
+/// created by a [State] that implements [TickerProvider],
+/// and then further animations may be derived from it using
+/// e.g. [Tween] or [CurvedAnimation].
+/// The animations might be used to configure an [AnimatedWidget]
+/// (using one of its many subclasses like [FadeTransition]),
+/// or their values might be used directly.
+///
+/// For example, the [AnimationController] may represent
+/// the abstract progress of the animation from 0.0 to 1.0;
+/// then a [CurvedAnimation] might apply an easing curve;
+/// and a [SizeTween] and [ColorTween] might each be applied to that
+/// to produce an [Animation<Size>] and an [Animation<Color>] that control
+/// a widget shrinking and changing color as the animation proceeds.
+///
+/// Because the [Animation] keeps the same identity as the animation proceeds,
+/// it provides a convenient way for a [StatefulWidget] that orchestrates
+/// a complex animation to communicate the animation's progress to its
+/// various child widgets.  Consider having higher-level widgets in the tree
+/// pass lower-level widgets the [Animation] itself, rather than its value,
+/// in order to avoid rebuilding the higher-level widgets on each frame
+/// even while the animation is active.
```

Comment:
```suggestion
/// while the animation is active.
```

Reply:
I don't feel strongly on this line but I think the "even while" version is slightly clearer. Otherwise it reads equally well as if it's saying that this optimization avoids rebuilding those widgets on each frame while the animation is active, but when it's inactive *they still get rebuilt*. (As opposed to the intended reading, where when it's inactive *it wouldn't be causing rebuilds anyway*.)

That other reading doesn't ultimately make sense in context. But the reader may be confused for a bit because the docs seem to be saying something that doesn't make sense, so that the reader has to sort out if the error is in their understanding of how things work, or in the docs, or (as it would turn out in this case) that they've hit an unintended reading of this sentence of the docs and there's a different reading that does make sense.

Reply:
After re-reading, I agree—it's pretty cool to only rebuild leaf widgets while there's an active animation!